commands/init.go

```diff
@@ -37,14 +39,32 @@ func transformInitArgs(args *Args) error {
 		return nil
 	}
 
-	name, err := utils.DirName()
-	if err != nil {
-		return err
+	var (
+		name   string
+		newDir bool
+		err    error
+	)
+
+	if args.IsParamsEmpty() {
+		name, err = utils.DirName()
+		if err != nil {
+			return err
+		}
+	} else {
+		name = args.LastParam()
+		newDir = true
 	}
 
 	project := github.NewProject("", name, "")
 	url := project.GitURL("", "", true)
-	args.After("git", "remote", "add", "origin", url)
+
+	cmds := []string{"git"}
+	if newDir {
+		cmds = append(cmds, "--git-dir", filepath.Join(name, ".git"))
```

Comment:
`git init --separate-git-dir` creates a filesytem-agnostic symlink to current folder or a folder passed in. For both cases, we need to run `git-remote` in the current folder or the folder passed in, instead of the folder as part of the `--separate-git-dir` flag which contains the bare repo:

```
$ git init foo --separate-git-dir=../bar
$ cd foo
$ cat .git
gitdir: /Users/owenou/src/bar
```